src/content/structured/get-started/testing-with-react-testing-library.mdx

```diff
@@ -0,0 +1,232 @@
+---
+path: "/get-started/install-components/testing-with-react-testing-library"
+
+navPriority: 8
+
+date: "2023-01-20"
+
+title: "Testing with React Testing Library"
+
+subTitle: "How to test components with React Testing Library."
+
+contribute: "https://github.com/mi6/ic-design-system/tree/main/src/content/structured/get-started/testing-with-react-testing-library.mdx"
+---
+
+## About React Testing Library
+
+[React Testing Library (RTL)](https://testing-library.com/docs/react-testing-library/intro) provides a set of utility functions which interrogate the DOM tree in order to build maintainable React tests. The recommended approach with React Testing Library is to test React components based on how the users view the components rather than the implementation of the component. For example, a user would view a button by the label 'Add', so the library provides a method called `getByText()` to facilitate that.
+
+## Shadow DOM Testing Library
+
+It is not possible to exclusively use RTL to create React tests which include `@ukic/react` components, as `@ukic/react` components consist of React-wrapped web components which use the [shadow DOM](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_shadow_DOM) (`<ShadowRoot>`). RTL does not provide utility functions which traverse beyond the shadow DOM. 
```

Comment:
```suggestion
It is not possible to exclusively use RTL to test `@ukic/react` components, as `@ukic/react` components are React-wrapped web components which use the [shadow DOM](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_shadow_DOM) (`<ShadowRoot>`). RTL does not provide utility functions which traverse beyond the shadow DOM. 
```

This was a long sentence so I've tried to simplify it

Reply:
Thanks. I was going nose blind with this paragraph